Microsoft.Toolkit.Uwp.UI.Controls/MasterDetailsView/MasterDetailsView.cs

```diff
@@ -324,23 +294,33 @@ private void UpdateViewState()
                 ViewState = MasterDetailsViewState.Both;
             }
 
-            var after = ViewState;
-
-            if (before != after)
+            if (previousState != ViewState)
             {
-                ViewStateChanged?.Invoke(this, after);
+                ViewStateChanged?.Invoke(this, ViewState);
+                SetBackButtonVisibility(previousState);
```

Comment:
Why is the back button visibility being set with the old state and not the current state?

Reply:
The function is already using the current state to decide if it needs to change the visibility. The previous state is used to decide if it needs to restore the back button to what it was before the states changed.

Reply:
ping @skendrot. Can you review this one more time so we can get it in before our code freeze end of day today ?